libraries/freertos_plus/aws/ota/include/aws_iot_ota_agent.h

```diff
@@ -451,7 +371,75 @@ typedef int16_t (* pxOTAPALWriteBlockCallback_t)( OTA_FileContext_t * const C,
 typedef OTA_JobParseErr_t (* pxOTACustomJobCallback_t)( const char * pcJSON,
                                                         uint32_t ulMsgLen );
 
-/* OTA PAL callback structure */
+
+/*--------------------------- OTA structs ----------------------------*/
+
+/**
+ * @structs{ota,OTA library}
+ */
+
+/* A composite cryptographic signature structure able to hold our largest supported signature. */
+
+#define kOTA_MaxSignatureSize    256        /* Max bytes supported for a file signature (2048 bit RSA is 256 bytes). */
+
+typedef struct
+{
+    uint16_t usSize;                         /* Size, in bytes, of the signature. */
+    uint8_t ucData[ kOTA_MaxSignatureSize ]; /* The binary signature data. */
```

Comment:
This means that we're always using 256 bytes of storage, even though our majority use case is ECDSA. I've typically seen this situation better handled by defining a zero-length array as the last element in the typedef.